### DISCUSSION TITLE
Is InputLeap compatible with Barrier?

### DISCUSSION THREAD

**Discussion by ViBE-HU** [input-leap/input-leap#2054] [Q&A]
I have a few machines which has Barrier on it and starts with the system automatically. On the server machine I installed InputLeap and it instantly connected to the Barrier client. Not clear is it an issue or a feature. But as I see InputLeap use the Barrier's configuration and log file. Is this normal (planned)?

**Comment by nbolton** [comment]
If you're talking about network, should be compatible. The protocol didn't change between Barrier and Input Leap AFAIK.

**Comment by ViBE-HU** [comment]
Cool. But in this case is it accidental that the program use Barrier's files?

**Reply by nbolton** [reply]
> Cool. But in this case is it accidental that the program use Barrier's files?

I'm not sure what you mean. Are you asking if a Barrier server/client can work with an Input Leap server/client? If so, the issue is purely network related (where there is no issue because the protocol is the same).

**Reply by ViBE-HU** [reply]
No. I mean it is not clear why InputLeap using directly the Barrier configuration file and logs files. FYI: I am still using Barrier and following InputLeap since the guys started the project. And Barrier is configured on every machine. But I noticed that InputLeap instantly started with my Barrier settings and there was no installation wizard/setup process at first start (dunno is it removed or just because of the existing Barrier configuration the program skipped it). So it just "imported" everything from Barrier without asking it.

**Reply by nbolton** [reply]
Ah, looks like I misunderstood. So you're asking if you can have both Barrier and Input Leap installed at the same time running side by side on the same computer?

**Reply by ViBE-HU** [reply]
No. That is already happening :D InputLeap accessing and using(?) Barrier files and settings instead of it's own. And it doing without any user permission. Atm it is not clear what are the goals of InputLeap, but compatibility is not equal with replacing. And as far as I understand InputLeap is not a continuation of Barrier, but an alternative which may compatible with it.

So to sum up: I'm still not sure about this behaviour since the README does not clarifies the compatibility. Back in the days there was a conversation here or maybe on irc about the backward compatibility and my last memory was that it is not a goal, but a possibility till the code not change significantly.

**Reply by shymega** [reply]
This is a known bug.

**Reply by ViBE-HU** [reply]
> This is a known bug.

Thanks for the clarification.

**Comment by p12tic** [comment]
Input Leap takes Barrier configuration data and migrates to the Input Leap format on the first run. After the first run old Barrier configuration data is ignored.
